### ISSUE TITLE
Loading a saved game/map with locked monsters feature oddities.

### ISSUE THREAD

**Issue by GuntherDW** [svkaiser/Doom64EX#80]
I noticed that in the current version on git (at least on Linux?) there appear to be some oddities when you save and/or load a game in conjunction with the "Lock monsters" feature in the features folder.

This also happens when changing maps in general (when the feature was turned on before).

It appears you cannot disable the feature until you actively either quit the game, or go to the main menu to start a new game, or load an unaffected save game.

At that point, you'll no longer be able to turn on  the "Lock monsters" feature at all. on/off doesn't affect the game at this point.
Loading another saved game again with it turned on however again "locks" them in the "frozen" position.

I've also got it to segfault like this at one point, but was unsuccessful to get it to segfault again later.

Basically it appears the feature is getting set somewhere it shouldn't and in effect getting "stuck" somehow.

- Versions
 Doom64EX : 8953438d348ee4f4e08fc60243aa0d19ff83afae

**Comment by dotfloat** [comment]
Thanks a bunch!

So the relevant flags were only set when a "brand new" session was loaded (ie. not when restarting a map or advancing to the next one). I changed it so that setting `sv_lockmonsters` and others (`p_allowjump` etc) sets the relevant game flags immediately.

As for the segfault. Did it happen when exiting to the main menu? Seems like there is an invalid access when doing so.

**Comment by GuntherDW** [comment]
I believe it was indeed. But seeing that it was a random crash and I haven't been able to trigger it again I just shrugged it off as a random crash.
